### PR TITLE
Improving tests

### PR DESCRIPTION
**Pull request by jonasvdd**
- [x] Use selenium on more use-cases
- [x] Check whether callbacks are effectively made
  - [x] Verify whether the content is correct
- [x] Test compatibility with non-scatter figure objects

Still TODO:
- [ ] check the `reset_axes` command to see whether there is data in each subplot

### CODE REVIEW

**Comment by jonasvdd** [comment]
@jvdd

File: plotly_resampler/figure_resampler.py [hunk 1]
```diff
@@ -134,6 +135,12 @@ def check_update_trace_data(self, trace, start=None, end=None):
             The end index for which we want the resampled data to be updated to,
             by default None
 
+        Returns
+        -------
+        Optional[Union[dict, BaseTraceType]]
+            If the matching hf_series is found in hf_dict, a(n updated) trace will be
```
Comment (jvdd) [inline plotly_resampler/figure_resampler.py:141]:
```suggestion
            If the matching hf_series is found in hf_dict, an (updated) trace will be
```

File: plotly_resampler/figure_resampler.py [hunk 2]
```diff
@@ -448,6 +469,10 @@ def add_trace(
                 if isinstance(hf_hovertext, np.ndarray):
                     hf_hovertext = hf_hovertext[not_nan_mask]
 
+            # if the categorical or string-like hf-y data is send to
```
Comment (jvdd) [inline plotly_resampler/figure_resampler.py:472]:
```suggestion
            # if the categorical or string-like hf_y data is send to
```

**Comment by jonasvdd** [comment]
@jvdd 😴

**Pull request merged**